### PR TITLE
specs: 4844 fault-proof preimage key type specs

### PR DESCRIPTION
**Pull request by protolambda**
Draft specs for EIP-4844 support in the fault-proof preimage oracle.

### CODE REVIEW

File: specs/fault-proof.md [hunk 1]
```diff
@@ -129,7 +131,31 @@ This enables fault proof programs to adopt any new pre-image schemes without VM
 It is up to the user to index the special pre-image values by this key scheme,
 as there is no way to revert it to the original commitment without knowing said commitment or value.
 
-#### Type `4-128`: reserved range
+#### Type `4`: Global SHA2-256 key
+
+A SHA-256 pre-image.
+
+Key: the SHA-256 hash, with the first byte overwritten with the type byte: `4 ++ sha256(data)`.
```
Comment (tynes) [inline specs/fault-proof.md]: does this need the slice `[1:]`?

**Review by clabby — APPROVED** [review]
LGTM, pending @tynes' nit

File: specs/fault-proof.md (hunk 1 above)
Comment (protolambda) [inline specs/fault-proof.md]:
```suggestion
Key: the SHA-256 hash, with the first byte overwritten with the type byte: `4 ++ sha256(data)[1:]`.
```

**Comment by protolambda** [comment]
Will re-open this PR into new specs repo. See #9121

**Comment by ajit2903** [comment]
This pull request updates the documentation for the fault proof pre-image key types to add support for new key types and clarify the reserved ranges. The changes ensure the spec is up-to-date with recent protocol developments, particularly around SHA-256 and EIP-4844 blob point-evaluation support.

**Pre-image key type additions and documentation updates:**

* Added documentation for Type 4 (Global SHA2-256 key) and Type 5 (Global EIP-4844 Point-evaluation key), specifying their usage and key formats. [[1]](diffhunk://#diff-a99f3028f4f185213498f966cf3c5990e4087979a4bc00db85616814c5d7679eL14-R16) [[2]](diffhunk://#diff-a99f3028f4f185213498f966cf3c5990e4087979a4bc00db85616814c5d7679eL132-R158)
* Adjusted the reserved range to start from Type 6 instead of Type 4, reflecting the new key types. [[1]](diffhunk://#diff-a99f3028f4f185213498f966cf3c5990e4087979a4bc00db85616814c5d7679eL14-R16) [[2]](diffhunk://#diff-a99f3028f4f185213498f966cf3c5990e4087979a4bc00db85616814c5d7679eL132-R158)

**Pull request closed without merging**